ingeniamotion/information.py

```diff
@@ -132,3 +142,261 @@ def register_exists(
+        """Return dictionary categories instance.
+
+        Args:
+            alias: Drive alias.
+
+        Returns:
+            Categories instance.
+        """
+        drive = self.mc.servos[alias]
+        dictionary_categories = drive.dictionary.categories
+        category_ids = dictionary_categories.category_ids
+        categories: dict[str, str] = {}
+        for cat_id in category_ids:
+            categories[cat_id] = dictionary_categories.labels(cat_id)["en_US"]
+        return categories
+
+    def get_dictionary_file_name(self, alias: str) -> str:
+        """Return dictionary file name.
+
+        Args:
+            alias: Drive alias.
+
+        Returns:
+            Dictionary file name.
+        """
+        drive = self.mc.servos[alias]
+        return str(os.path.basename(drive.dictionary.path))
+
+    def get_encoded_image_from_dictionary(self, alias: str) -> Optional[str]:
+        """Get the encoded product image from a drive dictionary.
+
+        This function reads a dictionary of a drive, and it parses whether the dictionary file has a
+        DriveImage tag and its content.
+
+        Args:
+            alias: Alias of the drive.
+
+        Returns:
+            The encoded image or NoneType object.
+        """
+        drive = self.mc.servos[alias]
+        # Read encoded image in XDF dictionary file
+        dictionary_path = drive.dictionary.path
+        try:
+            with open(dictionary_path, "r", encoding="utf-8") as xdf_file:
+                tree = ET.parse(xdf_file)
+        except FileNotFoundError:
+            raise FileNotFoundError(f"There is not any xdf file in the path: {dictionary_path}")
+        root = tree.getroot()
+        try:
+            image_element = root.findall(f"./DriveImage")
+            if image_element[0].text is not None and image_element[0].text.strip():
+                return f"{image_element[0].text}"
+            else:
+                # If the content in DriveImage tag is empty
+                return None
+        except IndexError:
+            # If there is no DriveImage tag in dictionary file
+            return None
+
+    def get_drive_info_coco_moco(
+        self, alias: str
+    ) -> tuple[list[Optional[int]], list[Optional[int]], list[Optional[str]], list[Optional[int]]]:
+        """Get info from COCO and MOCO registers.
+
+        Args:
+            alias: Servo alias.
+
+        Returns:
+            Product codes (COCO, MOCO).
+            Revision numbers (COCO, MOCO).
+            FW versions (COCO, MOCO).
+            Serial numbers (COCO, MOCO).
+
+        """
+        prod_codes: list[Optional[int]] = [None, None]
+        rev_numbers: list[Optional[int]] = [None, None]
+        fw_versions: list[Optional[str]] = [None, None]
+        serial_number: list[Optional[int]] = [None, None]
+
+        for subnode in [0, 1]:
+            # Product codes
+            try:
+                prod_codes[subnode] = self.mc.communication.get_register(
```

Comment:
Info module should not have functions that read registers, move it to communication or configuration